src/OneExplorer/OneExplorer.ts

```diff
@@ -631,6 +632,69 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<Node> {
     vscode.commands.executeCommand('revealFileInOS', node.uri);
   }
 
+  /**
+   * Rename a file of all types of nodes (baseModel, product, config) excepts for directory.
+   * It only alters the file name, not the path.
+   * @command one.explorer.rename
+   */
+  rename(node: Node): void {
+    // TODO: prohibit special characters for security ('..', '*', etc)
+    let warningMessage;
+
+    if (node.type === NodeType.baseModel) {
+      // TODO automatically change the corresponding files
+      warningMessage = `WARNING! ${
+          node.getChildren()
+              .map(node => `'${node.name}'`)
+              .toString()} will disappear from the view.`;
+    } else if (node.type === NodeType.product) {
+      // TODO automatically change the corresponding files
+      warningMessage = `WARNING! '${node.name}' may disappear from the view.`;
+    } else if (node.type === NodeType.config) {
+      // DO NOTHING
+      // Renaming config doesn't affect ONE Explorer view
+    } else if (node.type === NodeType.directory) {
+      assert.fail('Renaming is not allowed for directories.');
+    }
```

Comment:
(optional) It would be better to use switch statement for readability and for removing multiple if statement.
```suggestion
  switch(node.type) {
    case NodeType.baseModel: {
      // TODO automatically change the corresponding files
      warningMessage = `WARNING! ${
          node.getChildren()
              .map(node => `'${node.name}'`)
              .toString()} will disappear from the view.`;
    }
    case NodeType.product: {
      // TODO automatically change the corresponding files
      warningMessage = `WARNING! '${node.name}' may disappear from the view.`;
    }
    case NodeType.config: {
      // DO NOTHING
      // Renaming config doesn't affect ONE Explorer view
    }
    case NodeType.directory: {
      assert.fail('Renaming is not allowed for directories.');
    }
    default: {
      // ..
    }
  }
```

Reply:
Cool, thanks :-D